syntax = "proto3";

package live;

message PriceLevel {
  double price = 1;
  double volume = 2;
}

message LiveSnapshot {
  string source = 1;        // BINANCE
  string symbol = 2;        // BTCUSDT
  string exchange_ts = 3;   // Binance event time
  string ingest_ts = 4;     // Ingestor receive time

  repeated PriceLevel bids = 5;
  repeated PriceLevel asks = 6;
  double mid_price = 7;
}

service LiveFeedService {
  rpc StreamSnapshots (SubscribeRequest) returns (stream LiveSnapshot);
}

message SubscribeRequest {
  string source = 1;  // BINANCE
}
